container/openejb-core/src/main/java/org/apache/openejb/assembler/classic/Assembler.java

```diff
@@ -2469,36 +2469,40 @@ public void destroyApplication(final AppInfo appInfo) throws UndeployException {
     }
 
     private void destroyLookedUpResource(final Context globalContext, final String id, final String name) throws NamingException {
+        // check to see if the resource is a LazyObjectReference that has not been initialized
+        // if it is, we'll remove the LazyObjectReference, rather than do a lookup causing it
+        // to be instantiated
+        final String ctx = name.substring(0, name.lastIndexOf("/"));
```

Comment:
I'd like to see more safety here, that lastIndexOf isn't -1 etc. I know it wasn't there before, but just seems to be worth now you're on it.